WIP : Unit testing for custom status feature

manojmalik20 opened the pull request:
Added unit tests for custom status emoji component
Added unit tests for custom status text component
Added unit tests for clear button component
Added unit tests for custom status suggestion component
Added unit tests for custom status modal component
Updated unit tests for settings_sidebar
Updated unit tests for drawer item

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A brief description of what this pull request does.
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

#### Checklist
<!--
Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.
-->
- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates

#### Device Information
This PR was tested on: <!-- Device name(s), OS version(s) --> 

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs (for both iOS and Android if possible).
-->

manojmalik20 committed:
WIP : Unit testing for custom status feature
Added unit tests for custom status emoji component
Added unit tests for custom status text component
Added unit tests for clear button component
Added unit tests for custom status suggestion component
Added unit tests for custom status modal component
Updated unit tests for settings_sidebar
Updated unit tests for drawer item

chetanyakan commented on app/components/custom_status/custom_status_emoji.test.tsx:
```diff
@@ -0,0 +1,71 @@
+import {shallow} from 'enzyme';
+import React from 'react';
+
+import configureStore from 'redux-mock-store';
```
```suggestion
import configureStore from 'test/test_store';
```

chetanyakan commented on package.json:
```diff
@@ -112,6 +112,7 @@
     "@types/react-native-video": "5.0.3",
     "@types/react-redux": "7.1.15",
     "@types/react-test-renderer": "16.9.3",
+    "@types/redux-mock-store": "1.0.2",
```
Remove this dependency.

chetanyakan commented on package-lock.json:
```diff
@@ -9401,6 +9401,15 @@
         "@types/react": "*"
```
Revert any changes to this file

manojmalik20 committed:
Review fixes

manojmalik20 committed:
Updated tests
Updated snapshots of post header
Added tests and updated snapshtots in user profile screen
